splinepy/helpme/fit.py

```diff
@@ -127,7 +127,9 @@ def compute_knot_vector(degree, n_control_points, u_k, n_fitting_points):
         for j in range(1, n_control_points - degree):
             i = int(j * d)
             alpha = (j * d) - i
-            knot_vector[j + degree] = (1 - alpha) * u_k[i - 1] + alpha * u_k[i]
+            knot_vector[j + degree] = (1 - alpha) * u_k[
+                i - 1, 0
+            ] + alpha * u_k[i, 0]
```

Comment:
Are we sure that in all cases there is always just one element in the array? Otherwise, we would need to be more careful in how we change this line.

Reply:
Why is u_k 2D? any special reason for that?

Reply:
`u_k` is only 2D (n x 2) in the `surface`-fitting method and 1D (n x 1) in `curve`-fitting.
`compute_knot_vector` is then called with `u_k` in `curve` and `curve` is called with `u_k = u_k[0]` / `u_k = u_k[1]` for each dimension in `surface`.

Maybe we have to change the call of `curve` in the `surface` module to `u_k = u_k[:, 0]` /`u_k = u_k[:, 1]` instead of `u_k = u_k[0]` / `u_k = u_k[1]` because this seems unclear?

But in general, `u_k` should only be 1D in `compute_knot_vectors`, that's why I didn't add a second index.

Reply:
even nx1 is a 2D array. Alternatively we can `ravel()` to accommodate both cases

Reply:
yeah, at least within this function, you always expect `(n, 1)` arrays, so `ravel()`at the beginning (after shape check) should be fine

Reply:
nice flat